x/staking/keeper/delegation.go

```diff
@@ -60,19 +60,70 @@ func (k Keeper) GetAllDelegations(ctx sdk.Context) (delegations []types.Delegati
 func (k Keeper) GetValidatorDelegations(ctx sdk.Context, valAddr sdk.ValAddress) (delegations []types.Delegation) {
 	store := ctx.KVStore(k.storeKey)
 
-	iterator := storetypes.KVStorePrefixIterator(store, types.DelegationKey)
+	iterator := storetypes.KVStorePrefixIterator(store, types.GetDelegationsByValPrefixKey(valAddr))
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
-		delegation := types.MustUnmarshalDelegation(k.cdc, iterator.Value())
-		if delegation.GetValidatorAddr().Equals(valAddr) {
-			delegations = append(delegations, delegation)
+		var delegation types.Delegation
+		valAddr, delAddr, err := types.ParseDelegationsByValKey(iterator.Key())
+		if err != nil {
+			panic(err)
+		}
+
+		bz := store.Get(types.GetDelegationKey(delAddr, valAddr))
+		if err := k.cdc.Unmarshal(bz, &delegation); err != nil {
+			panic(err)
 		}
+
+		delegations = append(delegations, delegation)
 	}
 
 	return delegations
 }
 
+func (k Keeper) UpdateValidatorDelegations(ctx sdk.Context, existingValAddr, newValAddr sdk.ValAddress) {
```

Comment:
When and where is this method called? Is it only in benchmarks?

Reply:
> Is it only in benchmarks?
Right now, yes it is, but this can be used for updating operating address in follow-up PR.

Reply:
done, moved to test file.